ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -143,20 +144,26 @@ def _map_sample(self, sample, exclude_fields):
             rendered = jinja2.Template(
                 self.template, undefined=jinja2.StrictUndefined
             ).render(**filtered_sample, json_dump=json.dumps, text_proc=text_proc)
-
-        except jinja2.TemplateError as e:
-            error_message = str(e)
-            if "The formatted text is empty." in error_message:
+        except Exception as e:
+            if isinstance(e, text_proc.GarbageUtteranceError):
                 print("Formatted text is empty. Setting output to None.")
                 sample[self.new_column_name] = None
                 self.template_failures += 1
-            else:
+                return sample
+            elif isinstance(e, text_proc.EmptyTranscriptError):
+                print("Empty transcript after processing. Setting output to None.")
+                sample[self.new_column_name] = None
+                self.template_failures += 1
```

Comment:
The name `template_failures` is a bit confusing (considering the actual `jinja2.TemplateError` in line 158).